docs(js): Rework SvelteKit manual setup into quick start guide (#13632)
getsentry/sentry-docs: docs/platforms/javascript/guides/sveltekit/manual-setup.mdx

---
title: Manual Setup
sidebar_order: 1
description: "Learn how to manually set up Sentry in your SvelteKit app and capture your first errors."
---

<Alert type="info">
  For the fastest setup, we recommend using the [wizard
  installer](/platforms/javascript/guides/sveltekit).
</Alert>

## Prerequisites

You need:

- A Sentry [account](https://sentry.io/signup/) and [project](/product/projects/)
- Your application up and running

<Expandable title="Notes on SvelteKit adapter compatibility">
The SvelteKit Sentry SDK is designed to work out of the box with the following SvelteKit adapters:

- [Adapter-auto](https://kit.svelte.dev/docs/adapter-auto) – for Vercel; other platforms might work but we don't guarantee compatibility at this time

> - [Adapter-vercel](https://kit.svelte.dev/docs/adapter-vercel) – only for Node (Lambda) runtimes, not yet Vercel's edge runtime

Reviewer: ```suggestion
- [Adapter-vercel](https://kit.svelte.dev/docs/adapter-vercel) – only for Node (Lambda) runtimes, Vercel's edge runtime isn't supported yet
```

- [Adapter-cloudflare](https://kit.svelte.dev/docs/adapter-cloudflare) – supported but requires [additional setup](https://docs.sentry.io/platforms/javascript/guides/cloudflare/frameworks/sveltekit/)
- [Adapter-node](https://kit.svelte.dev/docs/adapter-node)

Other adapters may work but aren't currently supported. We're looking into extending first-class support to more adapters in the future.

> Also, Sentry's SvelteKit SDK does not yet work with all non-node server runtimes, such as Vercel's edge runtime.

Reviewer: I'm having a little trouble with this sentence, are we saying that only runtimes based on Node.JS will work?

Author: I've updated the structure to make it clearer what's supported and what's not -- @chargome @Lms24 please check if I got it right :)

---
title: Manual Setup
sidebar_order: 1

> description: "Learn how to manually set up Sentry in your SvelteKit app and capture your first errors."

Reviewer 2: 😂 😭  damn

If you're updating your Sentry SDK to the latest version, check out our [migration guide](https://github.com/getsentry/sentry-javascript/blob/master/MIGRATION.md) to learn more about breaking changes.

## Step 2: Configure

You need to initialize and configure the Sentry SDK in three places: the client side, the server side, and your Vite config.

> ### Configure Client-side Sentry

Reviewer: ```suggestion
### Configure Client-Side Sentry
```

> Create a client hooks file `src/hooks.client.(js|ts)` if you don't have one already. In this file, import and initialize the Sentry SDK and add the `handleErrorWithSentry` function to the [`handleError` hook](https://svelte.dev/docs/kit/hooks#Shared-hooks-handleError).

Reviewer: ```suggestion
Create a client hooks file `src/hooks.client.(js|ts)` in the root folder of your project if you don't have one already. In this file, import and initialize the Sentry SDK and add the `handleErrorWithSentry` function to the [`handleError` hook](https://svelte.dev/docs/kit/hooks#Shared-hooks-handleError).
```

Taking a guess here, but let's specify where this should live.

// export const handleError = handleErrorWithSentry();
```

> ### Configure Server-side Sentry

Reviewer: ```suggestion
### Configure Server-Side Sentry
```

// export const handleError = handleErrorWithSentry();

export const handle = Sentry.sentryHandle();
// Or use `sequence` if you're using your own handler(s):
// export const handle = sequence(Sentry.sentryHandle(), yourHandler());
```

<Alert level="warning" title="Important">
  For development, you can include your [Data Source
  Name](/concepts/key-terms/dsn-explainer/) (DSN) directly in these files.

>   However, in production, we strongly recommend you use an environment variable.

Reviewer: ```suggestion
  In production, we strongly recommend you use an environment variable for better security and flexibility.
```

Making a bit of a guess here again, but let's give a reason why.

// export const handleError = handleErrorWithSentry();

export const handle = Sentry.sentryHandle();
// Or use `sequence` if you're using your own handler(s):
// export const handle = sequence(Sentry.sentryHandle(), yourHandler());
```

<Alert level="warning" title="Important">
  For development, you can include your [Data Source
  Name](/concepts/key-terms/dsn-explainer/) (DSN) directly in these files.
  However, in production, we strongly recommend you use an environment variable
  to avoid accidentally exposing your DSN in the codebase.

> </Alert>

Author: Sorry about this! You're right, this does not make much sense, and I've removed the alert. 
I probably mixed this up with the Auth token and the Tip-Alert we have in the Next.js guide (there we recommend using the `NEXT_PUBLIC_SENTRY_DSN` public env var) 🙃

});
```

> ## Step 4: Cloudflare Pages Configuration (optional)

Reviewer: ```suggestion
## Step 4: Cloudflare Pages Configuration (Optional)
```

});
```

## Step 4: Cloudflare Pages Configuration (Optional)

If you're deploying your application to Cloudflare Pages, you need to adjust your server-side setup.
Follow this guide to [configure Sentry for Cloudflare](/platforms/javascript/guides/cloudflare/frameworks/sveltekit/).

> ## Step 5: Avoid Ad Blockers With Tunneling (Optional)

Reviewer 3: I'm not sure if we should show this section (yet). I saw we're doing this in NextJS but there, we have the NextJS-specific `tunnelRoute` option which basically eliminates the need to set up your own proxy server endpoint. For SvelteKit, this is necessary but unfortunately, our docs how to do this are terrible. They don't even show a JS example, which in SvelteKit (or any other JS meta framework), you'd most likely use. 

So my struggle is that this sounds like a logical step in the setup process but it'd block people quite a bit until they figure this out for JS and SvelteKit.

I think we have three options, out of which we should at least do (1) before we show this section in any guide:

1. Provide a basic JS example for a proxy, ideally one that works with compressed replay envelopes
2. Provide a proxy implementation from the SvelteKit SDK that users can plug into one of their sveltekit server routes
3. Provide a `tunnelRoute` equivalent for SvelteKit. This could work by leveraging the `handle` hook of SvelteKit (which we already do) and directly handling requests in there. Just a rough idea. 

So bottom line for this PR: Let's avoid showing this section for now. Does this sound reasonable?

Author: > Do we want to show tunneling in all manual setup guides eventually?
Yes! Unfortunately, I forgot to add tunneling to most QS guides (but I will soon update them so they're consistent :) ).

I'm not super happy with our solution here because users have to leave the page (I didn't want to include the instructions from the Troubleshooting page here because they're too complex for this guide). On the other hand, users need to be aware that this is a thing, especially if they wonder why their errors don't appear in Sentry (and that should not happen when they're in the quick start guide...).

> Side-note: I think long-term, we can do more here
>
> 1. Provide a proxy implementation from the SvelteKit SDK that users can plug into one of their sveltekit server routes
> 2. Provide a tunnelRoute equivalent for SvelteKit. This could work by leveraging the handle hook of SvelteKit (which we already do) and directly handling requests in there. Just a rough idea.

This sounds great to me -- the more straightforward we can make this setup, the better 💯

## Step 6: Verify Your Setup

Let's test your setup and confirm that Sentry is working correctly and sending data to your Sentry project.

### Issues

To verify that Sentry captures errors and creates issues in your Sentry project, create a test page, for example, at `src/routes/sentry-example/+page.svelte` with a button that throws an error when clicked:

```javascript {filename:+page.svelte}
<button
  type="button"
  onclick={() => {
    throw new Error("Sentry Example Frontend Error");
  }}
>
  Throw error
</button>
```

<OnboardingOption optionId="performance" hideForThisOption>
  Open the page `sentry-example` in a browser and click the button to trigger a
  frontend error.
</OnboardingOption>

<PlatformContent includePath="getting-started-browser-sandbox-warning" />

<OnboardingOption optionId="performance">
### Tracing
To test tracing, create a test API route like `src/routes/sentry-example/+server.(js|ts)`:

> ```javascript {filename:+server.(js|ts)}

Author: Yes, thank you, it looks much better now!

> ```javascript {filename:+server.(js|ts)}

Author: I updated it to `html` for the `+page.svelte` snippets and left `javascript` for `+server.(js|ts)` -> let me know if they all should use html

> ```javascript {filename:+server.(js|ts)}

Reviewer 3: that sounds perfect to me, thx!

export const GET = async () => {
  throw new Error("Sentry Example API Route Error");
};
```

Next, update your test button to call this route and throw an error if the response isn't `ok`:

> ```javascript {filename:+page.svelte}

Reviewer 3: same here:

```suggestion
```html {filename:+page.svelte}
```

<script>
  import * as Sentry from '@sentry/sveltekit';

  function getSentryData() {
    Sentry.startSpan(
      {
        name: 'Example Frontend Span',
        op: 'test'
      },
      async () => {
        const res = await fetch('/sentry-example');
        if (!res.ok) {
          throw new Error('Sentry Example Frontend Error');
        }
      }
    );
  }
</script>

> <button type="button" onclick={handleClick}>

Reviewer 2: handleClick is not defined in the snippet

Author: 🙈Thanks!
